src/components/QuickGrid/QuickGridHeader.tsx

```diff
@@ -1,89 +1,103 @@
 import * as React from 'react';
 import * as classNames from 'classnames';
-import { IQuickGridProps, IQuickGridState, GridColumn } from './QuickGrid.Props';
-import { autobind } from '../../utilities/autobind';
+import { IGridHeaderState, IGridHeaderProps } from './QuickGridHeader.Props';
+import { GridColumn, SortDirection } from './QuickGrid.Props';
+import { GroupByToolbar } from './GroupByToolbar';
+import { HeaderColumn } from './HeaderColumn';
 import { Grid, SortIndicator } from 'react-virtualized';
 const DraggableCore = require('react-draggable').DraggableCore;
-
+import * as _ from 'lodash';
 import './QuickGrid.scss';
-export interface IGridHeaderProps {
-    columns: Array<GridColumn>;
-    columnWidths: Array<number>;
-    onResize: (newColumnWidths) => void;
-    sortColumn?: string;
-    sortDirection?: 'ASC' | 'DESC';
-    onSort: (sortBy: string, sortDirection: string) => void;
-    className?: string;
-    width: number;
-    scrollLeft: any;
-}
-
-interface HeaderColumnProps {
-    onClick?: any;
-    onKeyDown?: any;
-}
 
-export interface IGridHeaderState {
-    columnWidths: Array<number>;
-}
-
-export class GridHeader extends React.Component<IGridHeaderProps, IGridHeaderState> {
+export class GridHeader extends React.PureComponent<IGridHeaderProps, IGridHeaderState> {
     private _headerGrid: any;
     private columnMinWidths: Array<number>;
-    constructor(props) {
+    constructor(props: IGridHeaderProps) {
         super(props);
         this.state = {
             columnWidths: props.columnWidths
         };
-        this.columnMinWidths = props.columns.map((col) => { return col.minWidth || 0; });
+        this.columnMinWidths = this.getColumnMinWidths(props.headerColumns);
+    }
+
+    getColumnMinWidths(columns) {
+        return columns.map((col) => { return col.minWidth || 0; });
     }
+
     componentWillReceiveProps(nextProps) {
-        this.setState({ ...this.state, columnWidths: nextProps.columnWidths });
+        if (!_.isEqual(nextProps.columnWidths, this.props.columnWidths)) {
```

Comment:
Lodash isEqual performs deep comparison, but it seems to me that we could use shallow comparison. It is faster and more inline with the React